### PR TITLE
FEATURE: SD-2625 - update the SDK Dashboard to add the embed button on KD

### PR DESCRIPTION
**Pull request by vannguyengd**
feat: add one more property for the ButtonBar component
one property was added to position the children's content for the ButtonBar component

JIRA: SD-2625

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended check plugins`                                | Dashboard plugins tests                                    |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11885/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13909/ : SUCCESS in 9m 56s

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/types.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
  */
 export interface IButtonBarProps {
     buttons?: React.ReactNode;
+    childContentPosition?: "left" | "right";
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/types.ts:16]: Describe this props

Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/types.ts:16]: Define default "left"

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/DefaultButtonBar.tsx [hunk 2]
```diff
@@ -15,17 +15,19 @@ export const DefaultButtonBar: React.FC<PropsWithChildren<IButtonBarProps>> = (p
         editButtonProps,
         saveAsNewButtonProps,
         shareButtonProps,
+        childContentPosition,
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/DefaultButtonBar.tsx:18]: define default "left" in spread

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/DefaultButtonBar.tsx [hunk 3]
```diff
@@ -15,17 +15,19 @@ export const DefaultButtonBar: React.FC<PropsWithChildren<IButtonBarProps>> = (p
...
     } = props;
 
     // TODO INE allow customization of buttons via getter from props
     return (
         <div className="dash-control-buttons">
-            {children}
+            {(!childContentPosition || childContentPosition === "left") && children}
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/DefaultButtonBar.tsx:24]: When default "left" is defined remove check for existence

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/types.ts (hunk 1 above)
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/types.ts:17]: I added the note for this new prop and the default value in the DefaultButtonBar component. Thanks!

**Review by ivanmjartan — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11888/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13912/ : SUCCESS in 10m 11s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3540

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul/56/ : FAILURE in 13m 01s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/396/ : FAILURE in 1m 49s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13936/ : SUCCESS in 13m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3669/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2512/ : SUCCESS in 11m 34s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/434/ : FAILURE in 22m 13s (non-voting)

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboard.spec.ts**|9|-|69.7090s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|48.9270s|
|✅|**dashboardDateRelated.spec.ts**|8|-|79.5340s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|20.7470s|
|✅|**attributeFilterButton.spec.ts**|13|-|63.5460s|
|✅|**headlineChartDrilling.spec.ts**|2|-|5.5480s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|10.1510s|
|✅|**dashboardDependentFilter.spec.ts**|8|-|108.2700s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|9.4290s|
|✅|**dashboardKPI.spec.ts**|8|-|183.1880s|
|✅|**drilling.spec.ts**|7(2 skipped)|-|51.2430s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|24.4100s|
|✅|**pivotTableSizing.spec.ts**|1|-|6.6460s|
|✅|**dateFilter.spec.ts**|9|-|18.3190s|
|✅|**dashboardFiltering.spec.ts**|18(2 skipped)|-|174.5150s|
|✅|**barChartDrilling.spec.ts**|4|-|9.0990s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|17.6970s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|9.4690s|
|✅|**dashboardDnD.spec.ts**|7|-|68.5070s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|9.2940s|
|✅|**attributeFilter.spec.ts**|1|-|10.5540s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/435/).

**Pull request merged**